src/Data/Auth/Providers/NoAuthProvider.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { GetAuthContext, defaultAuthContext } from "./AuthContext";
+
+export const NoAuthProvider: React.FC<React.PropsWithChildren> = ({
+  children,
+}) => {
+  return (
+    <GetAuthContext.Provider
+      value={{
+        ...defaultAuthContext,
+        getToken: () => {
+          return "No token";
```

Comment:
If you have refactored the file with the interface, you'll see that you won't need the override of the value here and can simply either return the `<GetAuthContext.Provider>{children}</GetAuthContext.Provider>`
or do that in the switch case and remove this file; choice is yours on that. I don't mind having it in a separate file for matter of cleaner cut.

Reply:
As far as I know it's not possible to remove assignment of the "value" property. the interface assigned to the context is correct, but still errors is there:
`Property 'value' is missing in type '{ children: ReactNode; }' but required in type 'ProviderProps<AuthContextInterface>'`

Reply:
I don't get why you need to override the defaults for no auth when no auth is the default. You can use the context as is in that case.
```ts
  const authContext= useContext(AuthContext);
  // rest of code
  return (
     <AuthContext.Provider value={authContext}>
         {children}
     </AuthContext.Provider>
  )
```